exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ResultRow.kt

```diff
@@ -19,16 +19,34 @@ class ResultRow(
      *
      * @see [getOrNull] to get null in the cases an exception would be thrown
      */
-    operator fun <T> get(expression: Expression<T>): T {
+    operator fun <T> get(expression: Expression<T>): T = getInternal(expression, checkNullability = true)
+
+    operator fun <T> set(expression: Expression<out T>, value: T) {
+        setInternal(expression, value)
+        lookUpCache.remove(expression)
+    }
+
+    private fun <T> setInternal(expression: Expression<out T>, value: T) {
+        val index = getExpressionIndex(expression)
+        data[index] = value
+    }
+
+    fun <T> hasValue(expression: Expression<T>): Boolean = fieldIndex[expression]?.let { data[it] != NotInitializedValue } ?: false
+
+    fun <T> getOrNull(expression: Expression<T>): T? = if (hasValue(expression)) getInternal(expression, checkNullability = false) else null
+
+    private fun <T> getInternal(expression: Expression<T>, checkNullability: Boolean): T {
         if (expression in lookUpCache) return lookUpCache[expression] as T
 
         val d = getRaw(expression)
 
-        if (d == null && expression is Column<*> && expression.dbDefaultValue != null && !expression.columnType.nullable) {
-            exposedLogger.warn(
-                "Column ${TransactionManager.current().fullIdentity(expression)} is marked as not null, " +
-                    "has default db value, but returns null. Possible have to re-read it from DB."
-            )
+        if (checkNullability) {
+            if (d == null && expression is Column<*> && expression.dbDefaultValue != null && ! expression.columnType.nullable) {
```

Comment:
Does this mean there's an issue with the Detekt check catching default formatting rule sets? `SpacingAroundUnaryOperator` in this case.

Reply:
It turns out that the `SpacingAroundUnaryOperator` was never enabled in the first place in our `detekt-config.yml` file, which is why it also passed [here](https://github.com/JetBrains/Exposed/blob/1afd391979eed949225c8fc3f309e6515991df32/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/SQLServerDialect.kt#L270C25-L270C140). It's active by default since [v1.22.0](SpacingAroundUnaryOperator), and we're using v1.21.0.

I confirmed this by modifying `SQLServerDialect.kt` in another PR and adding the rule to `detekt-config.yml`, and the detekt check failed.
<img width="1133" alt="Screenshot 2023-08-25 at 1 54 23 PM" src="https://github.com/JetBrains/Exposed/assets/38375996/f2b4ce16-ef9f-4c35-a277-2f20a9fc6e02">

I'll add the rule in a separate commit in this PR.